actionpack/lib/action_dispatch/http/mime_negotiation.rb

```diff
@@ -151,6 +154,17 @@ def valid_accept_header
       def use_accept_header
         !self.class.ignore_accept_header
       end
+
+      private
+
+      def extract_variant!
+        if parameters.key?(:format)
+          segments = parameters[:format].to_s.split("+")
```

Comment:
You means like this? (just rough implementation, tests are not passing) : https://github.com/FeGs/rails/commit/16f040c2a6b8559df652c66f231d089ab9064d94

This implementation allows `'post(.:format(+:variant))'` to be working, but variant is not assigned to `request.variant` yet.
```ruby
post.html+phone => { format: 'html+phone' } # before this patch applied
post.html+phone => { format: 'html', variant: 'phone' } # after this patch applied
```